### DISCUSSION TITLE
SunbirdEd-Portal :: Configuring quml player as web components in portal

### DISCUSSION THREAD

**Discussion by mithun30052001** [Sunbird-inQuiry/Community#105] [Bugs]
I followed the steps in the [documentation](https://github.com/project-sunbird/sunbird-quml-player?tab=readme-ov-file) to convert Quml Player into a web component.Iam getting the `"maxScore-undefined"` error due to the absence of null check in `sunbird-quml-player.js` in node_modules.

The error screen iam getting is as below:
<img width="1714" alt="Screenshot 2024-01-30 at 1 05 19 PM" src="https://github.com/Sunbird-inQuiry/Community/assets/81811776/24cbe96b-5c27-4eaa-b32f-7b616b4d5a74">

The null check i added in sunbird-quml-player.js  is the highlighted line in the below image:
<img width="961" alt="Screenshot 2024-01-30 at 1 07 07 PM" src="https://github.com/Sunbird-inQuiry/Community/assets/81811776/e8e93a91-dce5-4940-a648-b3addb21d411">

The outcome after the nullcheck:
![image (1)](https://github.com/Sunbird-inQuiry/Community/assets/81811776/a817f4af-b574-418c-b2d3-5ce79cea7e1a)

So kindly upgrade the library by adding the null check so that quml player could be integrated as web component in sunbird-ed portal

CC: [@vinukumar-vs](https://github.com/vinukumar-vs) [@rajeevsathish](https://github.com/rajeevsathish) [@princegupta1131](https://github.com/princegupta1131)

**Comment by rajnishdargan** [comment]
Hi @mithun30052001 

Could you please refer to the inQuiry microsite documentation for the integration of the web component?
Here is the link for using the [QuML player as a web component in the Angular app](https://inquiry.sunbird.org/use/developer-installation/question-set-player/installation#use-as-web-component-in-the-angular-app).
**Note:** The version of QuML Player expected to be used from release-6.0.0 of inQuiry is designed to work with Question/Questionset v2 endpoint APIs.

It is advisable not to use the [deprecated Quml-Player](https://inquiry.sunbird.org/use/deprecations/release-5.5.0#quml-player).
cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by mithun30052001** [reply]
Hi @rajnishdargan 

Thank you for the guidance. I already followed the instructions in the [microsite](https://inquiry.sunbird.org/use/developer-installation/question-set-player/installation#use-as-web-component-in-the-angular-app) for integrating the QuML player as a web component into sunbird-ed portal. However, I'm encountering the same "maxScore-undefined" error as mentioned above.

I would like to highlight that I'm using the latest version of the Quml-Web-Component library, which is version 3.0.2. I found this version on the official npm page: [Quml-Web-Component on npm](https://www.npmjs.com/package/@project-sunbird/sunbird-quml-player-web-component?activeTab=versions).

It would be really helpful if you had a version with this null check solved.

cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Comment by rajnishdargan** [comment]
Hi @mithun30052001,

Thank you for the update. We will investigate this issue and address it if necessary. Could you please provide some additional information?

1. In which release of SunbirdEd-portal are you attempting this integration?
2. Are you using the Questionset v2 API or the V1 API? It would be beneficial if you could share a screenshot of the network tab.
3. If you have submitted any draft PR for this integration, could you please share it with us? This will expedite the integration testing process.

cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131

**Reply by mithun30052001** [reply]
Hi @rajnishdargan 

Thank you for your response. Here are the details regarding the Quml web component integration:

 **Release of SunbirdEd-portal:**
        I am currently working on integrating the Quml web component into the SunbirdEd-portal version 7.0.0, planning to be implemented for the next release.

 **API version:**
        I am using the Questionset v2 API, specifically the api/question/v2/list endpoint for the integration.

<img width="846" alt="Screenshot 2024-01-31 at 1 46 19 PM" src="https://github.com/Sunbird-inQuiry/Community/assets/81811776/23137868-34e4-4dd2-9fcd-b2ab412343f8">


**Draft PR:**
        As of now, I haven't raised any pull requests related to this integration. However, I am actively working on the integration in the branch [all-players-as-web-components](https://github.com/mithun30052001/SunbirdEd-portal/tree/all-players-as-web-components).
        
cc: @sajeshkayyath @PoojaJManjooran @vinukumar-vs @rajeevsathish @princegupta1131
